unit_tests/test_provides.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2025 Canonical Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import charms_openstack.test_utils as test_utils
+
+from src import provides
+
+
+class TestRegisteredHooks(test_utils.TestRegisteredHooks):
+
+    def test_hooks(self):
+        defaults = []
+        hook_set = {
+            "when": {
+                "joined": ("endpoint.{endpoint_name}.joined",),
```

Comment:
Does this actually work?

Looking at other unit tests, they do use the expanded name in the test.

Reply:
The "provider endpoint" uses [the same flag format](https://github.com/mkalcok/charm-interface-cos-agent/blob/d9dc4e79cd8231b897feb31246a66816801d76b1/src/provides.py#L56) as this unit test expects, so it does work. Or was your question rather if the hooks work as expected in production?

Reply:
I just see that unit tests for other reactive interfaces make explicit use of the expected name in the unit test.

So the question is more if we get one less assertion, and can miss unintended changes to the endpoint names, or if we are smart and have one less thing to update in the event of future endpoint name change.

No strong opinions on this one.